backend/src/controllers/upload.ts

```diff
@@ -1,67 +1,82 @@
-import express from "express";
 import { Storage } from "@google-cloud/storage";
-import stream from "stream";
+import crypto from "crypto";
+import express from "express";
+import { StatusCodes } from "http-status-codes";
+import mime from "mime-types";
 import { Readable } from "stream";
+import { z } from "zod";
 import fullConfig from "../config/config";
-import crypto from "crypto";
 import { Upload } from "../models/upload";
 
+const MAX_UPLOAD_MB = 10;
+
 let storage = new Storage();
 
 if (fullConfig.googleSvcAcctKey !== "") {
-  storage = new Storage({keyFilename: fullConfig.googleSvcAcctKey});
+  storage = new Storage({ keyFilename: fullConfig.googleSvcAcctKey });
 }
 
-const storageBucketName = fullConfig.storageBucketName
+const storageBucketName = fullConfig.storageBucketName;
+
+const UploadRequestQuerySchema = z.object({
+  originalFileName: z.string(),
+  base64File: z
+    .string()
+    .max(MAX_UPLOAD_MB * 1024 * 1024, "File size too large"),
```

Comment:
This might cause actual max upload size to be around `MAX_UPLOAD_MB / 3` since each base 64 character represents 6 bits, but each javascript string character represents 2 bytes. so 16 bits vs 6 bits.